File: Seori/Weekly Contest 402/3184. Count Pairs That Form a Complete Day I.py
class Solution:
    def countCompleteDayPairs(self, hours: List[int]) -> int:
        # [1] 만약 두 시간의 합이 24로 나누어 떨어진다면, 두 시간은 하루를 완성한다. If the sum of two hours is divisible by 24, the two hours complete a day.
        answer = 0
        for i in range(len(hours)):
            for j in range(i + 1, len(hours)):
                if not (hours[i] + hours[j]) % 24:
                    answer += 1
        return answer

File: Lucy/2024/Weekly Contest 402/3184_Count_Pairs_That_Form_a_Complete_Day_I.js
/**
 * 3184. Count Pairs That Form a Complete Day I
 * URL: https://leetcode.com/problems/count-pairs-that-form-a-complete-day-i/
 * TOPIC:
 * DIFFICULTY: EASY
 * DATE: 2024.06.18(TUE)~
 */

/**
 * @param {number[]} hours
 * @return {number}
 */
var countCompleteDayPairs = function (hours) {
  const HOURS_LEN = hours.length;

  let completeDayCnt = 0;
  for (let i = 0; i < hours.length; i++) {
    for (let j = i + 1; j < hours.length; j++) {
      if ((hours[i] + hours[j]) % 24 === 0) completeDayCnt++;
    }
  }

  return completeDayCnt;
};

console.log(countCompleteDayPairs([12, 12, 30, 24, 24]));
